pkg/io/v2/providers/providers.go

```diff
@@ -0,0 +1,215 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package providers
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"path"
+	"strings"
+
+	"cloud.google.com/go/storage"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"gocloud.dev/blob"
+	_ "gocloud.dev/blob/fileblob"
+	"gocloud.dev/blob/gcsblob"
+	"gocloud.dev/blob/s3blob"
+	"gocloud.dev/gcp"
+	"golang.org/x/oauth2/google"
+)
+
+const (
+	// defaults to file because local paths were
+	// previously given without file:// prefix
+	defaultStorageProviderName = "file"
+
+	storageSeparator = "://"
+
+	providerFile = "file"
+	providerGS   = "gs"
+	providerS3   = "s3"
+)
+
+var storageProviders = []string{providerFile, providerGS, providerS3}
+
+// GetBucket opens and returns a gocloud blob.Bucket based on credentials and a path.
+// The path is used to discover which storageProvider should be used.
+//
+// If the storageProvider file is detected, we don't need any credentials and just open a file bucket
+// If no credentials are given, we just fall back to blob.OpenBucket which tries to auto discover credentials
+// e.g. via environment variables. For more details, see: https://gocloud.dev/howto/blob/
+//
+// If we specify credentials and an gs:// or s3:// path is used, credentials must be given in one of the
+// following formats:
+// * Google Cloud Storage (gs://):
```

Comment:
I think its enough if we say that is has to be a service account key, that format is standardized in contrast to the S3 stuff

Reply:
Done